tests/evd/test_evd.py

```diff
@@ -0,0 +1,441 @@
+    t = np.arange(0, time_series_length, acquisition_interval)
+    signal_phase = signal_rate * (t - t[0]) / 365.0
+    if k > 0:
+        seasonal = np.sin(2 * np.pi * k * t / 365.0) + np.cos(2 * np.pi * k * t / 365.0)
+        signal_phase = signal_phase + seasonal
+
+    # adding random temporal signal (which simulates atmosphere + DEM error + ...)
+    signal_phase = signal_phase + std_random * np.random.randn(len(t))
+    signal_phase = signal_phase - signal_phase[0]
+    # wrap the phase to -pi to p
+    signal_phase = np.angle(np.exp(1j * signal_phase))
+
+    return signal_phase, t
+
+
+def covarinace(C1, C2):
+
+    A1 = np.sum(np.abs(C1) ** 2)
+    A2 = np.sum(np.abs(C2) ** 2)
+
+    cov = np.sum(C1 * np.conjugate(C2)) / (np.sqrt(A1) * np.sqrt(A2))
+    return cov
+
+
+def compute_covariance_matrix(neighbor_stack):
+    numberOfSlc = neighbor_stack.shape[0]
+    cov_mat = np.zeros((numberOfSlc, numberOfSlc), dtype=np.complex64)
+    for ti in range(numberOfSlc):
+        for tj in range(ti + 1, numberOfSlc):
+            cov = covarinace(neighbor_stack[ti, :], neighbor_stack[tj, :])
+            cov_mat[ti, tj] = cov
+            cov_mat[tj, ti] = np.conjugate(cov)
+        cov_mat[ti, ti] = 1.0
+
+    return cov_mat
+
+
+def estimate_evd(cov_mat):
+
+    # estimate the wrapped phase based on the eigen value decomposition of the covariance matrix
+    w, v = np.linalg.eigh(cov_mat)
+
+    # the last eignevalue is the maximum eigenvalue
+    # However let's check to make sure
+    ind_max = np.argmax(w)
+
+    # the eignevector corresponding to the largest eigenvalue
+    # of the covariance matrix is the solution
+    evd_estimate = v[:, ind_max]
+
+    # refernce to the first acquisition
+    evd_estimate = evd_estimate * np.conjugate(evd_estimate[0])
+
+    return evd_estimate
+
+
+def simulate_bit_mask(Ny, Nx, filename="neighborhood_map"):
+
+    # number of uint32 bytes needed to store weights
+    number_of_bytes = np.ceil((Ny * Nx) / 32)
+
+    flags = np.ones((Ny, Nx), dtype=np.bool_)
+    flag_bits = np.zeros((Ny, Nx), dtype=np.uint8)
+
+    for ii in range(Ny):
+        for jj in range(Nx):
+            flag_bits[ii, jj] = flags[ii, jj].astype(np.uint8)
+
+    # create the weight dataset for 1 neighborhood
+    cols = Nx
+    rows = Ny
+    n_bands = int(number_of_bytes)
+    drv = gdal.GetDriverByName("ENVI")
+    options = ["INTERLEAVE=BIP"]
+    ds = drv.Create(filename, cols, rows, n_bands, gdal.GDT_UInt32, options)
+
+    half_window_y = int(Ny / 2)
+    half_window_x = int(Nx / 2)
+
+    ds.SetMetadata(
+        {"HALFWINDOWX": str(half_window_x), "HALFWINDOWY": str(half_window_y)}
+    )
+    ds = None
+
+    # above we created the ENVI hdr. Now let's write some data into the binary file
+
+    # Let's assume in the neighborhood of Nx*Ny all pixels are
+    # similar to the center pixel
+    s = ""
+    for ii in range(Nx * Ny * n_bands * 4 * 8):
+        s += "1"
```

Comment:
```suggestion
    s = "1" * (Nx * Ny * n_bands * 4 * 8)
```
Can skip explicit iteration.